lib/Service/StackService.php

```diff
@@ -365,4 +381,103 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId) {
+		if (is_numeric($id) === false) {
+			throw new BadRequestException('stack id must be a number');
+		}
+		if (is_numeric($boardId) === false) {
+			throw new BadRequestException('board id must be a number');
+		}
+
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_MANAGE);
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_READ);
+
+		if ($this->boardService->isArchived(null, $boardId)) {
+			throw new StatusException('Operation not allowed. This board is archived.');
+		}
+
+		$stack = $this->stackMapper->find($id);
+		$board = $this->boardMapper->find($boardId);
+
+		if ($stack->getBoardId() !== $board->getId()) {
+			throw new StatusException('Operation not allowed. Stack is not part of this board');
+		}
+
+
+		$newStack = new Stack();
+		$newStack->setTitle($stack->getTitle() . ' (' . $this->l10n->t('copy') . ')');
```

Comment:
We should only add the (copy) if the clone is going to the same board I guess, or ideally if there is another stack with the same title on the destination board.

Reply:
currently cloing is only possible on the same board